src/main/java/frc/robot/subsystems/climber/ClimberIOTalonFX.java

```diff
@@ -0,0 +1,225 @@
+package frc.robot.subsystems.climber;
+
+import static frc.robot.subsystems.climber.ClimberConstants.*;
+
+import com.ctre.phoenix6.BaseStatusSignal;
+import com.ctre.phoenix6.StatusCode;
+import com.ctre.phoenix6.StatusSignal;
+import com.ctre.phoenix6.configs.CurrentLimitsConfigs;
+import com.ctre.phoenix6.configs.MotionMagicConfigs;
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.controls.MotionMagicExpoTorqueCurrentFOC;
+import com.ctre.phoenix6.controls.TorqueCurrentFOC;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.hardware.core.CoreCANcoder;
+import com.ctre.phoenix6.signals.InvertedValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+import frc.lib.team3015.subsystem.FaultReporter;
+import frc.lib.team3061.RobotConfig;
+import frc.lib.team3061.drivetrain.swerve.Conversions;
+import frc.lib.team6328.util.Alert;
+import frc.lib.team6328.util.Alert.AlertType;
+import frc.lib.team6328.util.TunableNumber;
+
+/** TalonFX implementation of the generic SubsystemIO */
+public class ClimberIOTalonFX implements ClimberIO {
+  private TalonFX leftMotor;
+  private TalonFX rightMotor;
+
+  private CoreCANcoder leftEncoder;
+  private CoreCANcoder rightEncoder;
+
+  private MotionMagicExpoTorqueCurrentFOC leftPositionCurrentRequest;
+  private TorqueCurrentFOC leftSupplyAmpsRequest;
+
+  private MotionMagicExpoTorqueCurrentFOC rightPositionCurrentRequest;
+  private TorqueCurrentFOC rightSupplyAmpsRequest;
+
+  private StatusSignal<Double> leftVelocityRPSStatusSignal;
+  private StatusSignal<Double> leftPositionRotationsStatusSignal;
+  private StatusSignal<Double> leftStatorCurrentAmpsStatusSignal;
+  private StatusSignal<Double> leftSetpointStatusSignal;
+  private StatusSignal<Double> leftSupplyCurrentAmpsStatusSignal;
+
+  private StatusSignal<Double> rightVelocityRPSStatusSignal;
+  private StatusSignal<Double> rightPositionMetersStatusSignal;
```

Comment:
"Rotations" instead of "Meters" (we may convert to meters but this object returns values with units in rotations